src/main/java/com/epam/lowcost/DAO/UserDAOImpl.java

```diff
@@ -55,6 +58,16 @@ public User getById(long userId) {
         return user;
     }
 
+
+    @Override
+    public User findByEmail(String email) {
+        try {
+            return this.getJdbcTemplate().queryForObject("SELECT * FROM USERS WHERE email=?", UserRowMapper.userRowMapperInstance, email);
```

Comment:
call to field - bad practice. Try to create get method for your instance

Reply:
Made a getter for UserMapper instance